### PR TITLE
owner, metrics: add changefeed status in metrics and grafana

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 

Add metrics to monitor changefeed state #2251 

### What is changed and how it works?

* add `changefeedStatusGauge`
* create grafana panel for changefeed state.

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### Release note

- No release note

### CODE REVIEW

File: cdc/metrics_owner.go [hunk 1]
```diff
@@ -44,6 +44,13 @@ var (
 			Name:      "maintain_table_num",
 			Help:      "number of replicated tables maintained in owner",
 		}, []string{"changefeed", "capture", "type"})
+	changefeedStatusGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "ticdc",
+			Subsystem: "owner",
+			Name:      "status",
+			Help:      "changefeed status",
+		}, []string{"feedstate"})
```
Comment (overvenus) [inline cdc/metrics_owner.go:53]:
```suggestion
			Help:      "The status of changefeeds",
		}, []string{"changefeed", "state"})
```

We can record changefeed name and status in the gauge.

File: cdc/owner/metrics.go [hunk 2]
```diff
@@ -44,6 +44,13 @@ var (
 			Name:      "maintain_table_num",
 			Help:      "number of replicated tables maintained in owner",
 		}, []string{"changefeed", "capture", "type"})
+	changefeedStatusGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "ticdc",
```
Comment (ben1009) [inline cdc/owner/metrics.go:49]: seems the tag `ticdc`, `owner` should be const

**Review by ben1009 — COMMENTED** [review]
```
linting
  ✘  https://revive.run/r#exported  exported method FeedState.ToInt should have comment or be unexported  
  ./cdc/model/changefeed.go:55:1
```

File: metrics/grafana/ticdc.json [hunk 3]
```diff
@@ -106,9 +106,9 @@
       },
       {
         "datasource": "${DS_TEST-CLUSTER}",
-        "enable": false,
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:109]: This file changes so much, did you expand all tabs when exporting the metric file?

File: cdc/owner/metrics.go (hunk 2 above)
Comment (3AceShowHand) [reply cdc/owner/metrics.go:49]: only consider each `changefeed`, owner is not specified in this case.

File: cdc/model/changefeed.go [hunk 4]
```diff
@@ -52,6 +52,26 @@ const (
 	StateFinished FeedState = "finished"
 )
 
+// ToInt return a int for each `FeedState`
+func (s FeedState) ToInt() int {
```
Comment (overvenus) [inline cdc/model/changefeed.go:56]: Please add some comments, this method is only used in metrics.

File: metrics/grafana/ticdc.json [hunk 5]
```diff
@@ -106,9 +106,9 @@
...
+        "enable": true,
         "expr": "delta(tikv_cdc_region_resolve_status{status=\"resolved\"}[30s]) < BOOL -800",
-        "hide": false,
+        "hide": true,
```
Comment (overvenus) [inline metrics/grafana/ticdc.json:111]: Do not change unrelated lines.

**Review by overvenus — COMMENTED** [review]
You can import the latest ticdc.json before add new planes.

**Review by overvenus — APPROVED** [review]

File: cdc/owner/owner.go [hunk 6]
```diff
@@ -243,6 +244,7 @@ func (o *Owner) updateMetrics(state *model.GlobalReactorState) {
 			}
 			ownerMaintainTableNumGauge.WithLabelValues(changefeedID, captureInfo.AdvertiseAddr, maintainTableTypeTotal).Set(float64(len(taskStatus.Tables)))
 			ownerMaintainTableNumGauge.WithLabelValues(changefeedID, captureInfo.AdvertiseAddr, maintainTableTypeWip).Set(float64(len(taskStatus.Operation)))
+			changefeedStatusGauge.WithLabelValues(changefeedID).Set(float64(changefeedState.Info.State.ToInt()))
```
Comment (lonng) [inline cdc/owner/owner.go:247]: It's a disaster for others to write PromQL because they have no idea about the meaning of each number.

**Review by amyangfei — APPROVED** [review]

File: cdc/owner/metrics.go (hunk 2 above)
Comment (ben1009) [reply cdc/owner/metrics.go:49]: sorry, i mean the tag in
```
Namespace: "ticdc",
Subsystem: "owner",
```

**Review by ben1009 — APPROVED** [review]

**Comment by ben1009** [comment]
> /run-integration-tests
> /run-kafka-tests

does this mean we need manually trigger these two tests every pr, if yes, why not auto run, thanks

**Comment by amyangfei** [comment]
> > /run-integration-tests
> > /run-kafka-tests
> 
> does this mean we need manually trigger these two tests every pr, if yes, why not auto run, thanks

After `/merge`, all tests will be triggered. If part of the tests run with failure, we can re-trigger them by single command
